Microsoft.Toolkit.Uwp.SampleApp/Shell.xaml.cs

```diff
@@ -514,10 +520,105 @@ private void HamburgerMenu_OnOptionsItemClick(object sender, ItemClickEventArgs
             await Launcher.LaunchUriAsync(new Uri(e.Link));
         }
 
-        private void DocumentationTextblock_ImageResolving(object sender, ImageResolvingEventArgs e)
+        private async void DocumentationTextblock_ImageResolving(object sender, ImageResolvingEventArgs e)
         {
-            e.Image = new BitmapImage(new Uri("ms-appx:///Assets/pixel.png"));
-            e.Handled = true;
+            async Task<Stream> CopyStream(HttpContent source)
+            {
+                var stream = new MemoryStream();
+                await source.CopyToAsync(stream);
+                stream.Seek(0, SeekOrigin.Begin);
+                return stream;
+            }
+
+            var deferral = e.GetDeferral();
+            BitmapImage image = null;
+
+            var absolute = Uri.TryCreate(e.Url, UriKind.Absolute, out Uri Url);
+            if (!absolute)
+            {
+                Url = new Uri(documentationPath + e.Url);
+            }
+
+            if (Url.Scheme == "ms-appx")
+            {
+                image = new BitmapImage(Url);
+            }
+            else
+            {
+                IRandomAccessStream imageStream = null;
+                var localpath = $"{Url.Host}/{Url.LocalPath}";
+
+                // Cache only in Release
+#if !DEBUG
+                try
+                {
+                    imageStream = await StreamHelper.GetLocalCacheFileStreamAsync(localpath, Windows.Storage.FileAccessMode.Read);
+                }
+                catch
+                {
+                }
+#endif
+
+                if (imageStream == null)
+                {
+                    try
+                    {
+                        using (var client = new HttpClient())
```

Comment:
Keep a single static instance of HttpClient rather than creating a new one each time